runners/cli/src/main/kotlin/cli/main.kt

```diff
@@ -1,339 +1,344 @@
 package org.jetbrains.dokka
 
+import kotlinx.cli.*
 import org.jetbrains.dokka.DokkaConfiguration.ExternalDocumentationLink
 import org.jetbrains.dokka.utilities.DokkaConsoleLogger
-import java.io.File
-import java.io.FileNotFoundException
+import org.jetbrains.kotlin.utils.addToStdlib.cast
+import java.io.*
 import java.net.MalformedURLException
 import java.net.URL
-import java.net.URLClassLoader
 import java.nio.file.Files
 import java.nio.file.Paths
 
-open class GlobalArguments(parser: DokkaArgumentsParser) : DokkaConfiguration {
-    override val outputDir: String by parser.stringOption(
-        listOf("-output"),
-        "Output directory path",
-        ""
-    )
+class GlobalArguments(args: Array<String>) : DokkaConfiguration {
+
+    val parser = ArgParser("globalArguments", prefixStyle = ArgParser.OptionPrefixStyle.JVM)
+
+    override val outputDir by parser.option(ArgType.String, description = "Output directory path").required()
 
-    override val format: String by parser.stringOption(
-        listOf("-format"),
-        "Output format (text, html, gfm, jekyll, kotlin-website)",
-        ""
+    override val format by parser.option(
+        ArgType.String,
+        description = "Output format (text, html, gfm, jekyll, kotlin-website)"
+    ).required()
+
+    override val generateIndexPages by parser.option(ArgType.Boolean, description = "Generate index pages")
+        .default(false)
+
+    override val cacheRoot by parser.option(
+        ArgType.String,
+        description = "Path to cache folder, or 'default' to use ~/.cache/dokka, if not provided caching is disabled"
     )
 
-    override val pluginsClasspath: List<File> by parser.repeatableOption(
-        listOf("-dokkaPlugins"),
-        "List of jars with dokka plugins"
-    ) {
-        File(it)
-    }.also {
+    override val impliedPlatforms: List<String> = emptyList()
+
+    override val passesConfigurations by parser.option(
+        ArgTypeArgument,
+        description = "Single dokka pass",
+        fullName = "pass"
+    ).multiple()
+
+    override val pluginsConfiguration by parser.option(
+        ArgTypePlugin,
+        description = "Configuration for plugins in format fqPluginName=json^^fqPluginName=json..."
+    ).default(emptyMap())
+
+    override val pluginsClasspath by parser.option(
+        ArgTypeFile,
+        description = "List of jars with dokka plugins (allows many paths separated by the semicolon `;`)"
+    ).delimiter(";").also {
         Paths.get("./dokka-base.jar").toAbsolutePath().normalize().run {
-            if (Files.exists(this)) it.value.add(this.toFile())
+            if (Files.exists(this)) (it.value as MutableList<File>).add(this.toFile())
             else throw FileNotFoundException("Dokka base plugin is not found! Make sure you placed 'dokka-base.jar' containing base plugin along the cli jar file")
```

Comment:
I think we shouldn't throw an exception there as dokka-base is an optional jar. We should log it though

Reply:
There has been a task few months ago to link dokka-base to gradle/maven and enforce to provide jar in CLI, but can chagne